File: python/344_reverse_string.py
class Solution:
    def reverseString(self, s: list[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """ 
        for i in range(len(s)//2):
            first = s[i]
            last = s[len(s)-1-i]
            s[i] = last
            s[len(s)-1-i] = first
        
        print(s)

test = Solution()
test.reverseString(["H","a","n","n","a","h"])

File: typescript/src/344ReverseString.ts
function reverseString(s: string[]): void {
  for (let i = 0; i < s.length / 2; i++) {
    const first = s[i];
    const last = s[s.length - 1 - i];
    s[i] = last;
    s[s.length - 1 - i] = first;
  }
  console.log(s);
}
const s = ["h", "e", "l", "l", "o"];
reverseString(s);
